### PR TITLE
Updates to persistent org connections

### PR DESCRIPTION
**Pull request by davisagli**
# Critical Changes

# Changes
* There is now a default CumulusCI global connected app that can be used to connect to persistent orgs (assuming you know the credentials) without creating a new connected app. It's still possible to configure a custom connected app using `cci service connect connected_app` if more control over the connected app settings is needed.
* When CumulusCI is being run in a non-interactive context it can now obtain an access token for a persistent org using a JWT instead of a refresh token. This approach is used if the SFDX_CLIENT_ID and SFDX_HUB_KEY environment variables are set. This makes it easier to manage persistent org connections in the context of a hosted service because it's possible to replace the connected app's certificate without needing to obtain new refresh tokens for each org.

# Issues Closed

### CODE REVIEW

File: cumulusci/core/config/OrgConfig.py [hunk 1]
```diff
@@ -19,6 +22,23 @@ def __init__(self, config, name):
         super(OrgConfig, self).__init__(config)
 
     def refresh_oauth_token(self, keychain, connected_app=None):
+        SFDX_CLIENT_ID = os.environ.get("SFDX_CLIENT_ID")
+        SFDX_HUB_KEY = os.environ.get("SFDX_HUB_KEY")
+        if SFDX_CLIENT_ID and SFDX_HUB_KEY:
+            login_url = (
+                "https://test.salesforce.com"
```
Comment (Br4nd0R) [inline cumulusci/core/config/OrgConfig.py]: Would it make sense to have `SBX_LOGIN_URL` and `PROD_LOGIN_URL` attributes on `BaseConfig`?

Comment (davisagli) [reply cumulusci/core/config/OrgConfig.py]: I guess we could turn them into constants somewhere, and that could be helpful if someone wants to use CumulusCI with one of our internal test environments or when running the core application locally. I'd rather work on that in a separate PR when we can get input from someone who actually needs to do that, though.

File: cumulusci/core/keychain/BaseProjectKeychain.py [hunk 2]
```diff
@@ -11,6 +12,15 @@
 from cumulusci.core.sfdx import sfdx
 
 
+DEFAULT_CONNECTED_APP = ConnectedAppOAuthConfig(
+    {
+        "client_id": "3MVG9i1HRpGLXp.or6OVlWVWyn8DXi9xueKNM4npq_AWh.yqswojK9sE5WY7f.biP0w7bNJIENfXc7JMDZGO1",
```
Comment (Br4nd0R) [inline cumulusci/core/keychain/BaseProjectKeychain.py]: Hardcoding this is ok, as we don't expect to have to ever switch from the default connected app, correct?

Comment (davisagli) [reply cumulusci/core/keychain/BaseProjectKeychain.py]: That's the idea. If we did need to delete the existing connected app, users would need to upgrade to a new version of CumulusCI or configure their own connected app (the way they've needed to until now).

**Pull request merged**